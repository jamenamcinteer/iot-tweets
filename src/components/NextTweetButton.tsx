import React from "react"
import styled from "styled-components"
import { IPropsTheme } from "../interfaces/interfaces"

const Button = styled.button`
  appearance: none;
  background-color: transparent;
  border: 0;
  border-radius: 1.6rem;
  box-shadow: ${(props: IPropsTheme) => props.theme.boxShadowPrimary};
  padding: 1rem 1.5rem;
  color: ${(props: IPropsTheme) => props.theme.baseFontColor};
  cursor: pointer;
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeSmall};

  &:hover, &:focus {
    box-shadow: ${(props: IPropsTheme) => props.theme.boxShadowSecondary};
  }
`

const NextTweetButton = ({ clickHandler }) => {
  const onClick = (e) => {
    e.preventDefault()
    e.target.blur()
    clickHandler()
  }

  return (
    <Button onClick={onClick}>
      Another Tweet
      <svg viewBox="0 0 12 12" fill="none" style={{"width": "14px", "height": "14px", "marginLeft": "5px"}}>
        <path d="M6 0.666664L5.06 1.60666L8.78 5.33333H0.666668V6.66666H8.78L5.06 10.3933L6 11.3333L11.3333 6L6 0.666664Z" fill="white"/>
      </svg>
    </Button>
  )
}

export default NextTweetButton