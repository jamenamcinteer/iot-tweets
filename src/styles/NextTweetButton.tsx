import styled from "styled-components"
import { IPropsTheme } from "../interfaces/interfaces"

export const Button = styled.button`
  appearance: none;
  background-color: transparent;
  border: 0;
  border-radius: 1.6rem;
  box-shadow: ${(props: IPropsTheme) => props.theme.boxShadowPrimary};
  padding: 1rem 1.5rem;
  color: ${(props: IPropsTheme) => props.theme.baseFontColor};
  cursor: pointer;
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeSmall};
  font-family: ${(props: IPropsTheme) => props.theme.baseFontFamily};

  &:hover, &:focus {
    box-shadow: ${(props: IPropsTheme) => props.theme.boxShadowSecondary};
  }
`