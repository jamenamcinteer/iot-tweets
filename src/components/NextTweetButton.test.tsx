import React from 'react';
import { cleanup, render, fireEvent } from "@testing-library/react";
import "jest-styled-components";
import NextTweetButton from './NextTweetButton';

afterEach(cleanup);

it('renders as expected', () => {
  const clickHandler = jest.fn()

  const { asFragment } = render(<NextTweetButton clickHandler={clickHandler} />)
  expect(asFragment()).toMatchSnapshot()
})

it('should call clickHandler when clicked', () => {
  const clickHandler = jest.fn()

  const { getByText } = render(<NextTweetButton clickHandler={clickHandler} />)
  fireEvent.click(getByText('Another Tweet'))
  expect(clickHandler).toHaveBeenCalledTimes(1)
})