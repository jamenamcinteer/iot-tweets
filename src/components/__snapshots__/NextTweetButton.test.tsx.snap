// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  border: 0;
  border-radius: 1.6rem;
  padding: 1rem 1.5rem;
  cursor: pointer;
}

<button
    class="c0"
  >
    Another Tweet
    <svg
      fill="none"
      style="width: 14px; height: 14px; margin-left: 5px;"
      viewBox="0 0 12 12"
    >
      <path
        d="M6 0.666664L5.06 1.60666L8.78 5.33333H0.666668V6.66666H8.78L5.06 10.3933L6 11.3333L11.3333 6L6 0.666664Z"
        fill="white"
      />
    </svg>
  </button>
</DocumentFragment>
`;
